cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -186,6 +186,131 @@ _CCCL_HOST_DEVICE auto make_aligned_base_ptr(const T* ptr, int alignment) -> ali
   return aligned_base_ptr<T>{base_ptr, static_cast<int>(reinterpret_cast<const char*>(ptr) - base_ptr)};
 }
 
+// Our own version of ::cuda::aligned_size_t, since we cannot include <cuda/barrier> on CUDA_ARCH < 700
+template <_CUDA_VSTD::size_t _Alignment>
+struct aligned_size_t
+{
+  _CUDA_VSTD::size_t value; // TODO(bgruber): can this be an int?
+
+  _CCCL_HOST_DEVICE constexpr operator size_t() const
+  {
+    return value;
+  }
+};
+
+// TODO(bgruber): inline this as lambda in C++14
```

Comment:
We cannot use `cooperative_groups` with nvhpc stdpar, so this needs to be guarded by `!_CCCL_CUDA_COMPILER(NVHPC)`

Reply:
I tried replacing `cooperative_groups::memcpy_async` by `cuda::memcpy_async` and the former one just performs way better for `int8` and `int16`. It's almost a factor of 2 difference. CG tries to align to 16-byte copies by copying a few elements manually at the front and back, while `cuda::memcpy_async` does no such handling and just generates an unrolled loop copying bytes, which cripples performance.